tests/e2e/postback.spec.js

```diff
@@ -0,0 +1,67 @@
+context('Postback last transaction', () => {
+  describe('when you update a purchase with postback', () => {
+    let postback
+    let orderId
+
+    before(() => {
+      cy.configureCreditCard({ checkout: false })
+      cy.addProductToCart()
+      cy.goToCheckoutPage()
+      cy.fillCheckoutForm()
+      cy.selectCreditCard()
+      cy.fillCreditCardForm()
+      cy.placeOrder()
+    })
+
+    it('should be at order received page', () => {
+      cy.url({ timeout: 60000 })
+        .should('include', '/finalizar-compra/order-received/')
+      cy.contains('Pedido recebido')
+    })
+
+    it('should contain at least one postback', () => {
+      cy.task('pagarmejs:postback')
+        .then(postbacks => {
+          expect(postbacks).to.not.be.empty
+          postback = postbacks[0]
+          cy.getPayloadOrderNumber(postback.payload)
+            .then((id) => {
+              orderId = id
+            })
+        })
+    })
+
+    it('Postback URL should equals test domain', () => {
+      expect(postback).to.have
+        .property('request_url', 'http://woopagarme/wc-api/WC_Pagarme_Credit_Card_Gateway/')
+    })
+
+    it('should validate the current status of the order', () => {
+      cy.visit(`http://woopagarme/minha-conta/view-order/${orderId}/`)
+        .contains('atualmente está Aguardando.')
+    })
+
+    it('should update order transaction via postback', () => {
+      cy.request({
+          method: 'POST',
+          url: 'wc-api/WC_Pagarme_Credit_Card_Gateway/',
+          headers: JSON.parse(postback.headers),
+          body: postback.payload
+        })
+        .then((response) => {
+          expect(response.status).to.eq(200)
+          expect(response.body).to.be.empty
+        })
+    })
+
+    it('should validate the new status of the order', () => {
+      let status = {
```

Comment:
Por que é feita essa validação de status?

Reply:
Porque pegamos a última transação e não sabemos se ela será uma transação com falha ou não.

Contexto: no início do testes é feito um novo pedido, mas não consigo garantir que a transação retornada será essa ou de algum teste anterior, por isso a verificação para testar postback dos dois cenários de status.

Mantive o novo pedido no arquivo para garantir que não seja retornada a mesma transação caso os testes do arquivo seja executado separadamente mais de uma vez.